src/python/pants/backend/python/rules/hermetic_pex.py

```diff
@@ -1,64 +1,117 @@
-        # platforms, when we introduce platforms in https://github.com/pantsbuild/pants/issues/7735.
-        argv = ("python", pex_path, *pex_args)
-
-        hermetic_env = dict(
-            PATH=create_path_env_var(python_setup.interpreter_search_paths),
-            PEX_ROOT="./pex_root",
-            PEX_INHERIT_PATH="false",
-            PEX_IGNORE_RCFILES="true",
-            **subprocess_encoding_environment.invocation_environment_dict
-        )
-        if env:
-            hermetic_env.update(env)
+@dataclass(frozen=True)
+class HermeticPexRequest:
+    pex_path: str
+    exe_req: ExecuteProcessRequest
+
+
+class HermeticPex(Subsystem):
+    options_scope = "hermetic-pex-creation"
 
-        return ExecuteProcessRequest(
-            argv=argv, input_files=input_files, description=description, env=hermetic_env, **kwargs
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--python-executable-name",
+            type=str,
+            default="python",
+            help="The python executable to search for in --python-setup-interpreter-search-paths "
+            "when creating a pex with the v2 python backend.",
         )
+        register(
+            "--use-pex-cache",
+            type=bool,
+            default=False,
+            help="Whether to enable the pex local cache when executing hermetically.",
+        )
+
+    @property
+    def python_executable_name(self) -> str:
+        return cast(str, self.get_options().python_executable_name)
+
+    @property
+    def use_pex_cache(self) -> bool:
+        return cast(bool, self.get_options().use_pex_cache)
+
+    def caching_args(self) -> List[str]:
+        if self.use_pex_cache:
+            return []
+        return ["--disable-cache"]
+
+
+@rule
+def make_hermetic_pex_exe_request(
+    python_setup: PythonSetup,
+    subprocess_encoding_environment: SubprocessEncodingEnvironment,
+    downloaded_pex_bin: DownloadedPexBin,
+    hermetic_pex: HermeticPex,
+    req: HermeticPexRequest,
+) -> ExecuteProcessRequest:
+    """Creates an ExecuteProcessRequest that will run a PEX hermetically.
+
+    :param python_setup: The parameters for selecting python interpreters to use when invoking the
+                         PEX.
+    :param subprocess_encoding_environment: The locale settings to use for the PEX invocation.
+    :param pex_path: The path within `input_files` of the PEX file (or directory if a loose pex).
+    :param pex_args: The arguments to pass to the PEX executable.
+    :param description: A description of the process execution to be performed.
+    :param input_files: The files that contain the pex itself and any input files it needs to run
+                        against.
+    :param env: The environment to run the PEX in.
+    :param **kwargs: Any additional :class:`ExecuteProcessRequest` kwargs to pass through.
```

Comment:
This is stale and should be moved into the `create()` factory method I suggested above or be removed.